### PR TITLE
Handle unauthorized calls

### PR DESCRIPTION
**Pull request by allacherkes**

### CODE REVIEW

File: src/unleash_client_manager.ts [hunk 1]
```diff
@@ -9,13 +9,58 @@ import { registerStrategies } from "./strategy/appStudioStrategies";
...
+
+function handleUnauthorisedError(extensionName: string): void {
+  let num = initializeAttemptMap.get(extensionName)?.numAttempt;
+  num = (num || 0) + 1;
+  if (num < limitNumber) {
+    initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: 0, isBlocked: false });
+  }
+  if (num == limitNumber) {
+    initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: Date.now(), isBlocked: true });
+  }
```
Comment (AlexanderN8) [inline src/unleash_client_manager.ts:31]: I would try to change it to:
```js 
let isInitBlocked = num == limitNumber ? true : false;
initializeAttemptMap.set(extensionName, { numAttempt: num, timeAttempt: Date.now(), isBlocked: isInitBlocked });
```

File: src/unleash_client_manager.ts [hunk 2]
```diff
@@ -9,13 +9,58 @@ import { registerStrategies } from "./strategy/appStudioStrategies";
...
+}
+
+// handle unauthorised calls. Block calls for client initialization while in block time period.
+export function handleUnauthorisedCalls(extensionName: string, initializeAttemptMap: Map<string, InitializeAttempt>): void {
+  const initializationInfo = initializeAttemptMap.get(extensionName);
+  // check that client initialization will be not called while in block period
```
Comment (AlexanderN8) [inline src/unleash_client_manager.ts:37]: Please remove comments. Code should speak for himself.

**Review by AlexanderN8 — APPROVED** [review]

File: src/unleash_client_manager.ts (hunk 1 above)
Comment (allacherkes) [reply src/unleash_client_manager.ts]: Theoretically, I didnt want to modify this map  after limit is reached, but on the other side this code should not be reachable after limit is reached. Did this change. :)

File: src/unleash_client_manager.ts (hunk 2 above)
Comment (allacherkes) [reply src/unleash_client_manager.ts]: done

**Pull request merged**